workflows/methylation/methylation-cohort.wdl

```diff
@@ -0,0 +1,361 @@
+        memory: "~{memory_gb} GB"
+        cpu: 1
+        disks: "~{disk_size_gb} GB"
+        maxRetries: 1
+    }
+}
+
+task filter_probes {
+    meta {
+        description: "Filter probes based on standard deviation"
+        outputs: {
+            filtered_beta_values: "Filtered beta values for all samples",
+            filtered_probes: "Probes that were retained after filtering.",
+        }
+    }
+
+    parameter_meta {
+        beta_values: "Beta values for all samples"
+        num_probes: "Number of probes to retain after filtering"
+    }
+
+    input {
+        File beta_values
+        Int num_probes = 10000
+    }
+
+    Int disk_size_gb = ceil(size(beta_values, "GiB") * 2)
+
+    #@ except: LineWidth
+    command <<<
+        ln -s ~{beta_values} beta.csv
+
+        cat <<SCRIPT > filter.py
+        import csv
+        import pandas as pd
+        import numpy as np
+        import sys
+        import argparse
+
+        def get_args():
+            parser = argparse.ArgumentParser(
+                description="Filter probes based on standard deviation.")
+            parser.add_argument(
+                "--num_probes", type=int, default=10000, help="Number of probes to retain after filtering.")
+            parser.add_argument(
+                "beta", type=str, help="Beta values CSV file.")
+
+            args = parser.parse_args()
+
+            return args
+
+        if __name__ == "__main__":
+            args = get_args()
+
+            # Read beta values and compute standard deviation
+            data = []
+            with open("beta.csv", "r") as f:
+                reader = csv.reader(f)
+                header = next(reader)
+                for i, line in enumerate(reader):
+                    probe = line[0]
+                    sd = np.std([float(x) for x in line[1:]])
+                    data.append([probe, sd])
+
+            sd_df = pd.DataFrame(data, columns=["probe", "sd"]).set_index("probe")
+
+            # Filter probes based on standard deviation
+            filtered_probes = sd_df.sort_values("sd", ascending=False).head(args.num_probes).index
+
+            pd.Series(filtered_probes, index=filtered_probes).to_csv('filtered_probes.csv', index=False)
+
+            # Filter beta values
+            with open("beta.csv", "r") as f:
+                reader = csv.reader(f)
+                header = next(reader)
+                header[0] = "probe"
+                beta_data = [line for line in reader if line[0] in filtered_probes]
+
+            bd = pd.DataFrame(beta_data, columns=header).set_index("probe")
+            bd.to_csv("filtered_beta.csv")
+        SCRIPT
+
+        python filter.py --num_probes ~{num_probes} beta.csv
+    >>>
+
+    output {
+        File filtered_beta_values = "filtered_beta.csv"
+        File filtered_probes = "filtered_probes.csv"
+    }
+
+    #@ except: ContainerValue
+    runtime {
+        container: "quay.io/biocontainers/pandas:2.2.1"
+        memory: "28 GB"
```

Comment:
28 is an odd number. Isn't this going to depend on the size of the input?

Reply:
28 is a bit random. I'll take another look. It won't depend on the size of the input, but rather the width of the matrix, which is a bit harder to calculate upfront. It's the width because I'm reading a single row in to memory at a time.

Reply:
Speaking of the TSV functions, you could try reading into an `Array[Array[String]]` and getting the `length()` of the inner array?